### PR TITLE
Add keyrestore utility that can read backup bundles

### PR DESCRIPTION
**Pull request by csstaub**
Adds a new command, `keyrestore`, which can read a decrypted [Keywhiz backup bundle](https://github.com/square/keywhiz/pull/361) and install the files in the bundle into a secrets directory based on an existing keysync config. 

Example usage:
```
$ go build -o keyrestore ./cmd/keyrestore
$ ./keyrestore --config fixtures/configs/test-config.yaml --user test0 --group group0 --dir-name test fixtures/exportedSecretsBackupBundle.json
INFO[0000] Loading config                                file="fixtures/configs/test-config.yaml"
writing to fixtures/secrets/test/Hacking_Password
INFO[0000] Wrote file                                    dir="fixtures/secrets/test" file="Hacking_Password" logger="file_client"
WARN[0000] Secret permissions changed on disk            logger="file_client" secret="General_Password"
writing to fixtures/secrets/test/General_Password
INFO[0000] Wrote file                                    dir="fixtures/secrets/test" file="General_Password" logger="file_client"
$ ls -l fixtures/secrets/test
total 8
-r--r----- 1 cs 1846 6 Aug  4 15:52 General_Password
-r--r--r-- 1 cs 1846 4 Aug  4 15:52 Hacking_Password
```

Backup bundle example was generated with a dev Keywhiz instance:
```
curl -k -E fixtures/clients/client1.crt --key fixtures/clients/client1.key https://localhost:4444/automation/v2/backup/test/group/Blackops | gpg --decrypt
```

Note: must install a GPG key in the Keywhiz dev config for the backup export endpoint to work.

### CODE REVIEW

**Comment by mcpherrinm** [comment]
quick skim looks good; I'll review properly soon

File: .travis.yml [hunk 1]
```diff
@@ -8,5 +8,6 @@ env:
  - BUILD_GOOS=linux
 
 script:
- - GOOS=$BUILD_GOOS go build -o keysync cmd/keysync.go
+ - GOOS=$BUILD_GOOS go build -o keysync cmd/keysync/keysync.go
```
Comment (mcpherrinm) [inline .travis.yml]: please also update `build.sh` for this moved command

Comment (csstaub) [inline .travis.yml]: done

**Comment by mcpherrinm** [comment]
```./bundle.go:48: cannot use &client (type *BackupBundleClient) as type Client in return argument:
	*BackupBundleClient does not implement Client (wrong type for SecretList method)
		have SecretList() (map[string]Secret, bool)
		want SecretList() (map[string]Secret, error)```

this is a merge conflict from my previous PR, whoops

File: bundle.go [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+func (c BackupBundleClient) Secret(name string) (secret *Secret, err error) {
+	s, ok := c.secrets[name]
+	if !ok {
+		return nil, fmt.Errorf("unable to find %s in backup bundle", name)
+	}
+	return &s, nil
+}
+
+// SecretList returns all secrets in a bundle.
+func (c BackupBundleClient) SecretList() (map[string]Secret, bool) {
```
Comment (mcpherrinm) [inline bundle.go]: this needs to change to (map, error) and then `return c.secrets, nil`

Comment (csstaub) [inline bundle.go]: fixed

**Pull request merged**